File: _performance-golf/pg-ai/_gemini-cli/api-examples/java/src/test/java/com/example/gemini/CodeExecutionTest.java
/*
 * Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.example.gemini;

import com.google.genai.types.GenerateContentResponse;
import com.google.genai.types.Part;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertNotNull;


public class CodeExecutionTest {
    @Test
    public void test_codeExecutionBasic() {
        GenerateContentResponse response = assertDoesNotThrow(CodeExecution::codeExecutionBasic,
                "codeExecutionBasic returned an error");

        assertNotNull(response, "Response should not be null");

        List<Part> parts = response.candidates().get().getFirst().content().get().parts().get();
        assertNotNull(parts, "Response parts should not be null");
        assert !parts.isEmpty();

        assertNotNull(response.text(), "Response text should not be null");
        assertFalse(response.text().trim().isEmpty(), "Response text should not be empty");
    }

    @Test
    public void test_codeExecutionRequestOverride() {
        GenerateContentResponse response = assertDoesNotThrow(CodeExecution::codeExecutionRequestOverride,
                "codeExecutionRequestOverride returned an error");

        assertNotNull(response, "Response should not be null");

        assertNotNull(response.executableCode(), "ExecutableCode response should not be null");
        assertNotNull(response.codeExecutionResult(), "CodeExecutionResult response should not be null");
    }
}

File: _performance-golf/pg-ai/_gemini-cli/api-examples/javascript/code_execution.test.js
/**
 * @license
 * Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import assert from "node:assert";
import { test, describe } from "node:test";
import {
  codeExecutionBasic,
  codeExecutionRequestOverride,
  codeExecutionChat,
} from "./code_execution.js";

describe("code_execution", () => {
  test("codeExecutionBasic", async () => {
    const result = await codeExecutionBasic();
    // Check that the response contains parts and non-empty results.
    assert.ok(Array.isArray(result.parts) && result.parts.length > 0);
    assert.ok(result.text.length > 0);
  });

  test("codeExecutionRequestOverride", async () => {
    const result = await codeExecutionRequestOverride();
    // Check that the response contains non-empty results.
    assert.ok(result.executableCode.length > 0);
    assert.ok(result.codeExecutionResult.length > 0);
  });

  test("codeExecutionChat", async () => {
    const result = await codeExecutionChat();
    // Check that the response contains non-empty results.
    assert.ok(result.executableCode.length > 0);
    assert.ok(result.codeExecutionResult.length > 0);
  });
});
